src/topology/gene.rs

```diff
@@ -239,9 +239,9 @@ where
     T: Float,
 {
     fn cmp(&self, other: &Self) -> Ordering {
-        if self.output.layer == other.output.layer && self.output.index == other.output.index {
+        if self.output.layer == other.output.layer {
```

Comment:
This should stay the same `self.output.layer == other.output.layer && self.output.index == other.output.index`